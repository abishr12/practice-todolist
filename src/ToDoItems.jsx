import React from 'react';

const ToDoItems = ({entries, deleteItem}) => {
    
  let fullList = entries
    .map(entry => <li onClick={()=> deleteItem(entry.key)} key={entry.key}>{entry.task}</li>)

  return (
    <ul>
        {fullList}
    </ul>
  )
}

export default ToDoItems;