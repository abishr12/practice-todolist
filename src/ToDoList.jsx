import React, {Component} from 'react';
import ToDoItems from './ToDoItems';

class ToDoList extends Component {

  constructor(props) {
    super(props)
    this.state = {
      tasks: [],
      newTask: ''
    }

    this.handleChange = this
      .handleChange
      .bind(this);

    this.addItem = this
      .addItem
      .bind(this);

    this.deleteItem = this
      .deleteItem
      .bind(this);
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    })

  }

  deleteItem(key) {
    let {tasks} = this.state
    const filteredItems = tasks.filter(objKey => objKey === key)

    this.setState({tasks: filteredItems})

  }
  addItem(event) {
    event.preventDefault();
    let {newTask} = this.state

    let newTaskObj = {
      key: Date.now(),
      task: newTask
    };

    this.setState(prevState => ({
      tasks: prevState
        .tasks
        .concat(newTaskObj),
      newTask: ''
    }));

  }

  render() {
    return (
      <div>
        <form onSubmit={this.addItem}>
          <input
            onChange={this.handleChange}
            name='newTask'
            value={this.state.newTask}
            className="form-control"
            placeholder='Enter Task Here'/>
          <button className="btn btn-success">Add Task</button>
        </form>
        <ToDoItems deleteItem={this.deleteItem} entries={this.state.tasks}/>
      </div>
    )
  }
}

export default ToDoList;