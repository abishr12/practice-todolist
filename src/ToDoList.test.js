import React from 'react';
import ReactDOM from 'react-dom';
import ToDoList from './ToDoList.jsx';
import ToDoItems from './ToDoItems.jsx'
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()})
let dummyEntries = [{
    key: 11928,
    task: 'go to school'
}, {
    key: 119809,
    task: 'brush my teeth'
}
]

describe('<ToDoList/> shallow rendering', () =>{
    it('matches the snapshot', () => {
        const tree = shallow(<ToDoList />)
        expect(toJson(tree)).toMatchSnapshot()
    })
    it('correctly takes in input', () => {
        const wrapper = mount(<ToDoList />)
        const form = wrapper.find('form')
        form.simulate('submit', { target: { value: 'do some testing' } })
        expect(wrapper.state().tasks.length).toBe(1)
    })
    it('correctly removes input', () => {
        const wrapper = mount(<ToDoList />)
        const form = wrapper.find('form')
        form.simulate('submit', { target: { value: 'do some testing' } })
        const element = wrapper.find('li')
        element.simulate('click')
        form.simulate('submit', { target: { value: 'do some work' } })
        expect(wrapper.state().tasks.length).toBe(1)
    })
})

describe('<ToDoItems/> shallow rendering', () => {

    it('successfully takes in props', () => {
        const wrapper = shallow(<ToDoItems entries ={dummyEntries} />)
       
        expect(wrapper.getElement().props.children.length).toBe(2)
    })

    
})