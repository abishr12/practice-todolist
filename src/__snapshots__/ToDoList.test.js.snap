// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoList/> shallow rendering matches the snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="newTask"
      onChange={[Function]}
      placeholder="Enter Task Here"
    />
    <button
      className="btn btn-success"
    >
      Add Task
    </button>
  </form>
  <ToDoItems
    deleteItem={[Function]}
    entries={Array []}
  />
</div>
`;
